code/game/objects/items/devices/cictablet.dm

```diff
@@ -52,75 +40,58 @@
 	..()
 
 	if(src.allowed(user))
-		user.set_interaction(src)
-		interact(user)
+		tgui_interact(user)
 	else
 		to_chat(user, SPAN_DANGER("Access denied."))
 
-/obj/item/device/cotablet/interact(mob/user as mob)
-	if(!on)
-		return
 
-	user.set_interaction(src)
-	var/dat = "<body>"
-	if(announcement_faction != FACTION_MARINE && EvacuationAuthority.evac_status == EVACUATION_STATUS_INITIATING)
-		dat += "<B>Evacuation in Progress</B>\n<BR>\nETA: [EvacuationAuthority.get_status_panel_eta()]<BR>"
-
-	if(announcement_faction == FACTION_MARINE)
-		switch(state)
-			if(STATE_DEFAULT)
-				dat += "<BR><A HREF='?src=\ref[src];operation=announce'>Make an announcement</A>"
-				dat += "<BR><A HREF='?src=\ref[src];operation=award'>Award a medal</A>"
-				dat += "<BR><A HREF='?src=\ref[src];operation=mapview'>Tactical Map</A>"
-				dat += "<BR><hr>"
-				switch(EvacuationAuthority.evac_status)
-					if(EVACUATION_STATUS_STANDING_BY)
-						dat += "<BR><A HREF='?src=\ref[src];operation=evacuation_start'>Initiate Emergency Evacuation</A>"
-
-			if(STATE_EVACUATION)
-				dat += "Are you sure you want to evacuate the [MAIN_SHIP_NAME]? This cannot be undone from the tablet. <A HREF='?src=\ref[src];operation=evacuation_start'>Confirm</A>"
-	else
-		dat += "<BR><A HREF='?src=\ref[src];operation=announce'>Make an announcement</A>"
-		dat += "<BR><A HREF='?src=\ref[src];operation=mapview'>Tactical Map</A>"
+/obj/item/device/cotablet/ui_data(mob/user)
+	var/list/data = list()
 
-	dat += "<BR>[(state != STATE_DEFAULT) ? "<A HREF='?src=\ref[src];operation=main'>Main Menu</A>|" : ""]<A HREF='?src=\ref[user];mach_close=communications'>Close</A> "
-	show_browser(user, dat, tablet_name, "communications", "size=400x500")
-	onclose(user, "communications")
-	updateDialog()
+	data["faction"] = announcement_faction
+	data["alert_level"] = security_level
+	data["evac_status"] = EvacuationAuthority.evac_status
+	data["message_time_left"] = COOLDOWN_TIMELEFT(src, announcement_cooldown)
```

Comment:
dont do this, send the time it will expire at.
then the ui can count it down itself without needing backend updates